### PR TITLE
Update Android NDK to r12b.

### PR DESCRIPTION
**Pull request by chinmaygarde**
* Downloaded, repackaged and added the Linux and Mac NDK (r12b) to `gs://mojo`.
* Always use NDK platform level 24 on Android.
* Minor linker updates.

cc @jamesr @abarth @jason-simmons

### CODE REVIEW

File: build/config/android/config.gni [hunk 1]
```diff
@@ -94,20 +94,19 @@ if (is_android) {
...
-  _android64_api_level = 21
   x86_64_android_sysroot_subdir =
-      "platforms/android-${_android64_api_level}/arch-x86_64"
+      "platforms/android-${_android_api_level}/arch-x86_64"
   arm64_android_sysroot_subdir =
-      "platforms/android-${_android64_api_level}/arch-arm64"
+      "platforms/android-${_android_api_level}/arch-arm64"
   mips64_android_sysroot_subdir =
-      "platforms/android-${_android64_api_level}/arch-mips64"
+      "platforms/android-${_android_api_level}/arch-mips64"
```
Comment (jamesr) [inline build/config/android/config.gni]: do you actually support mips?

Comment (chinmaygarde) [inline build/config/android/config.gni]: We don't have any builders that build for MIPS currently. Though it should not be stretch to get one up and running.

File: build/config/android/config.gni [hunk 2]
```diff
@@ -94,20 +94,19 @@ if (is_android) {
 
   # Subdirectories inside android_ndk_root that contain the sysroot for the
   # associated platform.
-  _android_api_level = 16
+  _android_api_level = 24
```
Comment (jamesr) [inline build/config/android/config.gni]: I always get these confused, but is api_level the minimum level you can target or the maximum level you can use features from?  This is going from Jellybean (4.1.x) to Nougat (7.0)

Comment (chinmaygarde) [inline build/config/android/config.gni]: Generally, the Android NDK platform level is orthogonal to the Android API level. This line just means that the toolchain assumes headers and libraries from Android Nougat are present. We still need to do the due diligence to not use features that are absent on older versions on Android (like Vulkan, which we try to `dlopen` and fall back to OpenGL ES). Likewise, these builds will still work on newer versions of Android. So really, this does not enforce a maximum or minimum version compatibility.

Comment (chinmaygarde) [inline build/config/android/config.gni]: We usually want to use the latest available NDK platform (barring issues with the toolchain, etc.) and then depending on the features we use, update the AndroidManifest to gate incompatible devices.

**Comment by chinmaygarde** [comment]
This is tested and good to go. lgtm sauce?

**Comment by abarth** [comment]
LGTM

**Comment by abarth** [comment]
SAUCE APPLIED

**Comment by chinmaygarde** [comment]
Thanks!

**Pull request merged**